asreviewcontrib/datatools/dedup.py

```diff
@@ -1,38 +1,44 @@
 import argparse
 
+import numpy as np
 import pandas as pd
 
-from asreview.data import load_data
 
+def dedup(asdata, pid='doi'):
+    initial_length = len(asdata.df)
 
-def dedup(input_path, output_path=None):
-    # read data in ASReview data object
-    asdata = load_data(input_path)
+    if pid in asdata.df.columns:
+        # replace NaN values with empty string to prevent astype(str) from creating literal 'NaN' string
+        asdata.df[pid].fillna('', inplace=True)
+
+        # convert to string to support non-string types, strip whitespaces and replace empty strings with NaN values
+        s_pid = asdata.df[pid].astype(str).str.strip().replace("", np.nan)
+
+        # remove records based on duplicate PIDs
+        asdata.df = asdata.df[(~s_pid.duplicated()) | (s_pid.isnull())].reset_index(drop=True)
+    else:
+        print(f"Not using {pid} for deduplication because there is no such data.")
```

Comment:
```suggestion
```

Reply:
I moved it to the entrypoint, or should we remove the warning entirely?